frontend/src/pages/patients/PatientsPage.tsx

```diff
@@ -65,8 +74,111 @@ function patientAliasFilterWhereVariables(value: string): PatientAliasWhere[] {
   }
 }
 
-export const PatientsPage: React.FunctionComponent = (props) => {
+function addCDashToCMOId(cmoId: string): string {
+  return cmoId.length === 6 ? `C-${cmoId}` : cmoId;
+}
+
+export default function PatientsPage({ setUserEmail }: { setUserEmail: any }) {
   const params = useParams();
+  const childRef = useRef<any>(null);
+
+  const [phiEnabled, setPhiEnabled] = useState(false);
+  const [patientIdsTriplets, setPatientIdsTriplets] = useState<
+    PatientIdsTriplet[]
+  >([]);
+  const [patientsListColumns, setPatientsListColumns] =
+    useState(PatientsListColumns);
+  const [popupOpen, setPopupOpen] = useState(false);
+  const [showAlertModal, setShowAlertModal] = useState(false);
+
+  useEffect(() => {
+    function handleLogin(event: any) {
+      if (event.origin !== "http://localhost:4001") return;
+      setUserEmail(event.data);
+      setPopupOpen(false);
+      childRef.current.handleSearch();
+    }
+
+    if (popupOpen) {
+      window.addEventListener("message", handleLogin);
+      return () => {
+        window.removeEventListener("message", handleLogin);
+      };
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [popupOpen]);
+
+  useEffect(() => {
+    if (phiEnabled && patientIdsTriplets.length > 0) {
+      const colsWithMRNs = PatientsListColumns.map((column) => {
+        if (column.headerName === "Patient MRN") {
+          return {
+            ...column,
+            hide: false,
+            valueGetter: (params: any) => {
+              const cmoId = params.data.value;
+              const patientIdsTriplet = patientIdsTriplets.find(
+                (triplet) => addCDashToCMOId(triplet.cmoId) === cmoId
+              );
+              if (patientIdsTriplet) {
+                return patientIdsTriplet.ptMrn;
+              } else {
+                return "";
+              }
+            },
+          };
+        } else {
+          return column;
+        }
+      });
+      setPatientsListColumns(colsWithMRNs);
+    } else {
+      setPatientsListColumns(PatientsListColumns);
+    }
+  }, [phiEnabled, patientIdsTriplets]);
+
+  async function fetchPatientIdsTriplets(
+    patientMrns: string[]
+  ): Promise<string[]> {
+    try {
+      const response = await fetch("http://localhost:4001/mrn-search", {
+        method: "POST",
+        credentials: "include",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(patientMrns),
```

Comment:
not urgent, but we should switch to a library that wraps fetch.  Axios library seems to be very commonly used.

Reply:
this is another reason to just graphql, because it will make everything consistent, instead of having two ways of fetching and handling data.

Reply:
the drawback is we have to learn a little more about graphQL. But that's not so bad!